meson.build

```diff
@@ -0,0 +1,320 @@
+project('notcurses', ['c', 'cpp'],
+        default_options:
+          ['c_std=gnu11',
+           'cpp_std=gnu++17',
+           'default_library=both',
+           'c_args=-Wall -Wextra -W -Wshadow -Wformat -fexceptions',
+           'cpp_args=-Wall -Wextra -W -Wshadow -Wformat -fexceptions'
+          ],
+        version: '2.2.4',
+        license: 'Apache-2.0',
+        meson_version: '>=0.57.1')
+
+fs = import('fs')
+
+cc = meson.get_compiler('c')
+cxx = meson.get_compiler('cpp')
+
+deps = []
+
+add_global_arguments('-Og', language: 'cpp')
+add_global_arguments('-Og', language: 'c')
+add_global_arguments('-D_FORTIFY_SOURCE=2', language: 'cpp')
+add_global_arguments('-D_FORTIFY_SOURCE=2', language: 'c')
+
+multimedia_backend = get_option('use_multimedia')
+
+message('Requested multimedia backend: ' + multimedia_backend)
+message('Requested build type: ' + get_option('buildtype'))
+
+# Because everything and it's mother requires threads...
+threads_dep = dependency('threads')
+
+deps += threads_dep
+deps += dependency('readline')
+
+# Due to some distros (*cough cough* alpine *cough cough*), terminfo sometimes
+# gets sucked into curses. Accept either, and have mercy on us all.
+
+terminfo = dependency('tinfo', required : false)
+
+if not terminfo.found()
+    deps += dependency('ncursesw')
+else
+    deps += terminfo
+endif
+
+if multimedia_backend == 'ffmpeg'
+  deps += dependency('libavcodec')
+  deps += dependency('libavformat')
+  deps += dependency('libavutil')
+  deps += dependency('libswscale')
+elif multimedia_backend == 'oiio'
+  deps += dependency('OpenImageIO')
+endif
+
+deps += cc.find_library('m', required : false)
+
+if get_option('use_doctest')
+  deps += dependency('doctest')
+endif
+
+if not cc.has_header('uniwbrk.h')
+  error('Could not find uniwbrk.h from GNU libunistring. STOP.')
+else
+  deps += cc.find_library('unistring', required : true)
+endif
+
+if get_option('use_qrcodegen')
+  deps += cc.find_library('libqrcodegen', required : true)
+endif
+
+deps += cc.find_library('rt')
```

Comment:
This should have `required: false` for portability. Many platforms do not have `rt`, it is a part of libc instead.